### PR TITLE
Fix VS complaining about missing projects in slns

### PR DESCRIPTION
**Pull request by ViktorHofer**
Contributes to https://github.com/dotnet/runtime/issues/65552

These changes contribute to fixing the broken Visual Studio experience when interacting with out-of-band projects, like Microsoft.Extensions.* in solution files.

The changes are grouped by commits:
1. Relevant infrastructure changes to avoid references to inbox only libraries.
2. Update Microsoft.Extensions.* ref projects
3. Update Microsoft.Extensions.* src projects (just cleaning-up the property that doesn't need to be set anymore)
4. Update other out-of-band projects that are referenced by Microsoft.Extensions* projects. I plan to submit an extra PR for the remaining ones that aren't part of the Microsoft.Extensions.* graph as I wanted to keep that out of this PR to keep the change minimal.
5. Update two source files for which an analyzer complained about the missing usage of Span now that System.Memory.dll reaches the compiler.

cc @ericstj as we discussed this in our 1:1 last Thursday.

### CODE REVIEW

File: eng/targetingpacks.targets [hunk 1]
```diff
@@ -15,8 +15,7 @@
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(DisableImplicitFrameworkReferences)' != 'true' and
-                            '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
```
Comment (eerhardt) [inline eng/targetingpacks.targets:18]: I know we have no plans of ever introducing a new TargetFramworkIdentifier, but I think it is a good idea to still check that this is `.NETCoreApp` to future proof. Just checking the version might not always be the best approach.

File: eng/targetingpacks.targets [hunk 2]
```diff
@@ -71,16 +70,11 @@
   <Target Name="RemoveFrameworkReferences"
```
Comment (eerhardt) [inline eng/targetingpacks.targets:70]: The comment above this line:

```xml
<!-- DisableImplicitAssemblyReferences support. -->
```

Seems out of date now that this code no longer references `DisableImplicitAssemblyReferences`.

File: src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/Microsoft.Extensions.Hosting.WindowsServices.csproj [hunk 3]
```diff
@@ -3,12 +3,9 @@
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent);$(NetCoreAppMinimum);netstandard2.1;netstandard2.0;$(NetFrameworkMinimum)</TargetFrameworks>
     <EnableDefaultItems>true</EnableDefaultItems>
-    <!-- Use targeting pack references instead of granular ones in the project file. -->
-    <DisableImplicitAssemblyReferences>false</DisableImplicitAssemblyReferences>
     <IsPackable>true</IsPackable>
     <PackageDescription>.NET hosting infrastructure for Windows Services.</PackageDescription>
-	<AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <EnableLibraryImportGenerator>true</EnableLibraryImportGenerator>
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/Microsoft.Extensions.Hosting.WindowsServices.csproj:11]: Why are you removing `EnableLibraryImportGenerator`?

File: src/libraries/Microsoft.Win32.SystemEvents/src/Microsoft.Win32.SystemEvents.csproj [hunk 4]
```diff
@@ -1,12 +1,13 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
+    <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent);$(NetCoreAppMinimum)-windows;$(NetCoreAppMinimum);netstandard2.0;$(NetFrameworkMinimum)</TargetFrameworks>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <Nullable>enable</Nullable>
```
Comment (eerhardt) [inline src/libraries/Microsoft.Win32.SystemEvents/src/Microsoft.Win32.SystemEvents.csproj:5]:
```suggestion
```

I think this was a merge error.

File: src/libraries/Microsoft.Extensions.Logging.Console/src/Microsoft.Extensions.Logging.Console.csproj [hunk 5]
```diff
@@ -5,10 +5,7 @@
     <EnableDefaultItems>true</EnableDefaultItems>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <DefineConstants>$(DefineConstants);NO_SUPPRESS_GC_TRANSITION</DefineConstants>
-    <!-- Use targeting pack references instead of granular ones in the project file. -->
-    <DisableImplicitAssemblyReferences>false</DisableImplicitAssemblyReferences>
     <IncludePlatformAttributes>true</IncludePlatformAttributes>
-    <EnableLibraryImportGenerator>true</EnableLibraryImportGenerator>
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Logging.Console/src/Microsoft.Extensions.Logging.Console.csproj:11]: Similar question - why the removal here?

File: src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/Microsoft.Extensions.Hosting.WindowsServices.csproj (hunk 3 above)
Comment (ViktorHofer) [reply src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/Microsoft.Extensions.Hosting.WindowsServices.csproj:11]: That property is already set to true by the generators.targets file: https://github.com/dotnet/runtime/blob/3f4ac7f4c38d4e8043175639dd3a47f921500442/eng/generators.targets#L23-L29.

File: src/libraries/Microsoft.Extensions.Logging.Console/src/Microsoft.Extensions.Logging.Console.csproj (hunk 5 above)
Comment (ViktorHofer) [reply src/libraries/Microsoft.Extensions.Logging.Console/src/Microsoft.Extensions.Logging.Console.csproj:11]: same as above. They are already set by generators.targets.

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks for this cleanup!

**Review by ericstj — COMMENTED** [review]
This LGTM.  Seems like we should have docs somewhere that describe how to maintain references  in project files based on various criteria.  Do we need to update those?

**Comment by ViktorHofer** [comment]
We have project docs which are probably the right place to encode rules for references. I will submit a PR to update them.

**Pull request merged**